ee/components/stripe/Payment.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react";
+import { stringify } from "querystring";
+import { CardElement, useStripe, useElements } from "@stripe/react-stripe-js";
+import Button from "@components/ui/Button";
+import { useRouter } from "next/router";
+import useDarkMode from "@lib/core/browser/useDarkMode";
+import { PaymentData } from "@ee/lib/stripe/server";
+import { Prisma } from ".prisma/client";
+
+const CARD_OPTIONS = {
+  iconStyle: "solid" as const,
+  classes: {
+    base: "block p-2 w-full border-solid border-2 border-gray-300 rounded-md shadow-sm dark:bg-black dark:text-white dark:border-gray-900 focus-within:ring-black focus-within:border-black sm:text-sm",
+  },
+  style: {
+    base: {
+      color: "#000",
+      iconColor: "#000",
+      fontFamily: "ui-sans-serif, system-ui",
+      fontSmoothing: "antialiased",
+      fontSize: "16px",
+      "::placeholder": {
+        color: "#888888",
+      },
+    },
+  },
+};
+
+const eventTypeData = Prisma.validator<Prisma.EventTypeArgs>()({ select: { id: true } });
+type EventType = Prisma.EventTypeGetPayload<typeof eventTypeData>;
+
+const userData = Prisma.validator<Prisma.UserArgs>()({ select: { username: true } });
+type User = Prisma.UserGetPayload<typeof userData>;
+
+type Props = {
+  payment: {
+    data: PaymentData;
+  };
+  eventType: EventType;
+  user: User;
+};
+
+export default function PaymentComponent(props: Props) {
+  const router = useRouter();
+  const { name, date } = router.query;
+  const [error, setError] = useState<string | null>(null);
+  const [processing, setProcessing] = useState(false);
```

Comment:
explicit state would be cleaner -- example below -- could possibly replace `disabled` as well
```suggestion
type States =
  | {
      status: 'loading';
    }
  | {
      status: 'error';
      error: Error;
    }
  | {
      status: 'ok';
    };
const [state, setState] = useState<States>()
```